app/app.go

```diff
@@ -1468,6 +1475,28 @@ func (app *App) RegisterTendermintService(clientCtx client.Context) {
 	tmservice.RegisterTendermintService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.interfaceRegistry)
 }
 
+func (app *App) checkTotalBlockGasWanted(ctx sdk.Context, txs [][]byte) bool {
+	totalGasWanted := uint64(0)
+	for _, tx := range txs {
+		decoded, err := app.txDecoder(tx)
```

Comment:
Nit: If we have perf concern, we can make decode run in parallel, and then use atomic integer for calculating totalGasWanted

Reply:
yeah good point. I'll add it in in a later PR if we find this to be a bottleneck